io-sim-classes/README.md

```diff
@@ -0,0 +1,96 @@
+# Simulator Monad Class Hierarchy
+
+This package provides a monad class hierarchy which is an interface for both
+[io-sim](https://hackage.haskell.org/package/io-sim) and
+[IO](https://hackage.haskell.org/package/base-4.14.0.0/docs/GHC-IO.html#t:IO)
+monads.  It was developed with the following constraints in mind:
+
+* be a drop in replacement for `IO` monad;
+* `IO` instances does not alter its original semantics, providing a shallow
+  bindings to `async`, `base`, `stm` and `exception` packages;
+* provide zero cost abstractions.
+
+There are a few departures from this principles, usually visible in type
+signature, which we discuss in this document.  When using `IO`, for most of the
+interfaces, `GHC` can optimise away the provided abstractions with `-o1`
+optimisation level.
+
+## Exception Class Hierarchy
+
+This package provides an alternative class hierarchy giving access to
+exceptions api.  The `exception` package class hierarchy is also supported by
+`io-sim`, so you can also use either one.
+
+ The `MonadThrow` defined in this package allows to work with exceptions
+without having explicit access to `catch` or `mask`.  It only provides access
+to `throwIO`, `bracket`, `bracket_` and `finally`.  `MonadCatch` class provides
+api which allows to work with exceptions, e.g. `catch` or `bracketOnError`, and
```

Comment:
> `MonadCatch` class provides api which allows to work with exceptions,

That doesn't sound very fluent. I'm not a native speaker, but I'd go for something like "The `MonadCatch` class allows exceptions to be caught or handled".